tket/src/Transformations/Decomposition.cpp

```diff
@@ -563,6 +565,45 @@ static void best_exact_decomposition(
   }
 }
 
+// Given a 2-qb circuit with a single TK2, returns a tuple (pre, angles, post)
+// where pre/post are the parts of the circuit that come before/after the TK2
+// gate and angles are the 3 angles of the TK2 gate
+static auto get_TK2_angles(const Circuit &circ) {
```

Comment:
Since this function is essentially "undoing" what `CircPool::TK2_using_normalised_TK2()` did, it could perhaps be a bit more efficient if we refactored so that this function just took three angles, and was called both from this file and from the `CircPool` function. But I don't know if it is worth it.

Reply:
That was a very good idea. It made the code nicer.